### PR TITLE
Add utility script to overwrite RESTART fields from FV3-JEDI analysis

### LINKED ISSUE #8: Python utility to update analysis fields
Need to develop a python-based utility to update fields in FV3 RESTART files with those produced by FV3-JEDI.

### PR DESCRIPTION
**Pull request by CoryMartin-NOAA**
Closes #8 

This adds a utility python script to overwrite FV3 RESTART fields using arrays from a FV3-JEDI analysis file in the FV3 RESTART format.

### CODE REVIEW

**Review by kevindougherty-noaa — DISMISSED** [review]
Looks good to me logically.

File: UFS-DA/ush/update_restart.py [hunk 1]
```diff
@@ -0,0 +1,61 @@
...
+            continue
+        if ncv not in rst.variables:
+            raise KeyError(f"Variable {ncv} not in both files")
+        data = anl.variables[ncv][:]
+        outvar = rst.variables[ncv]
+        outvar[:] = data
+
+    # close the files and flush to disk
+    anl.close()
+    rst.close()
```
Comment (aerorahul) [inline UFS-DA/ush/update_restart.py]:
```suggestion
    with nc.Dataset(analysis, 'r') as anl, nc.Dataset(restart, 'a') as rst:
        # check that dimensions match
        check_dims(anl, rst)

        # loop through variables now
        for ncv in anl.variables:
            if ncv in ignore_vars:
                continue
            if ncv not in rst.variables:
                raise KeyError(f"Variable {ncv} not in both files")
                
            data = anl.variables[ncv][:]
            outvar = rst.variables[ncv]
            outvar[:] = data
```

**Review by aerorahul — APPROVED** [review]
looks good.

**Review by kevindougherty-noaa — APPROVED** [review]

**Review by RussTreadon-NOAA — APPROVED** [review]
**Functionall review**

The changes in PR #9 have already been merged into develop.   Nonetheless, it is good to document this functional review.   

Use `update_restart.py` to process operational C768L127 atmges and atmanl for 2021122106.   Use `compare_ncfile.py` prior to execution of `update_restart.py` to confirm that contents of two files differ.   Execute `update_restart.py` with contents of atmanl replacing contents of atmges.   Execute `compare_ncfile.py` again.   atmanl and atmges contain identical field values.

Repeat for operational gdas sfc*nc files.   `update_restart.py` works as designed.   Repeat tests using operational enkfgdas (C384L127) atm and sfc nc files.   `update_restart.py` works as designed.

Timings follow (obtained via `time ./update_restart.py file1.nc file2.nc`)
-  C768L127:  13m37.794s
- C768L127 sfc*nc:  2m35.597s
- C384L127 atm*nc:  2m32.389s
- C384L127 sfc*nc:  0m18.899s

While the script works as designed, future updates should explore ways to reduce execution time.

**Comment by CoryMartin-NOAA** [comment]
@RussTreadon-NOAA thanks for timing tests. Wow, those are slow. These are for the Gaussian files and not the RESTART tile files? For our workflow we will be using the RESTART files and can use cfp to parallelize, but I think you've already shown that we are going to need a different approach when this becomes operational.

**Comment by RussTreadon-NOAA** [comment]
Correct, the timings are for gaussian grid files.    Here are timings to
execute update_restart.py for tile1 from
$COMROOTp3/gfs/prod/gdas.20211221/06/atmos/RESTART/ for the indicated
filetypes:

   - fv_core.res.tile1.nc:  1m10.675s
   - fv_tracer.res.tile1.nc:  1m5.698s

These timings are significantly faster.  This makes sense since tiles are
being processed.

**Comment by CoryMartin-NOAA** [comment]
Thanks @RussTreadon-NOAA , I bet a big part of that runtime for the Gaussian grid files is because of the netCDF compression.

**Comment by RussTreadon-NOAA** [comment]
I agree.  Compression costs cpu cycles.

**Pull request merged**